sdk/data_kusto/src/connection_string.rs

```diff
@@ -0,0 +1,389 @@
+    pub password: Option<&'a str>,
+    /// A String value that provides the application client ID to use when authenticating.
+    pub application_client_id: Option<&'a str>,
+    /// A String value that provides the application key to use when authenticating using an application secret flow.
+    pub application_key: Option<&'a str>,
+    /// A String value that instructs the client to perform application authenticating with the specified bearer token.
+    pub application_token: Option<&'a str>,
+    /// ...
+    pub application_certificate: Option<&'a str>,
+    /// A String value that provides the thumbprint of the client
+    /// certificate to use when using an application client certificate authenticating flow.
+    pub application_certificate_thumbprint: Option<&'a str>,
+    /// A String value that provides the name or ID of the tenant in which the application is registered.
+    pub authority_id: Option<&'a str>,
+    /// Denotes if MSI authorization should be used
+    pub msi_auth: Option<bool>,
+    pub msi_params: Option<&'a str>,
+    pub az_cli: Option<bool>,
+}
+
+impl<'a> PartialEq for ConnectionString<'a> {
+    fn eq(&self, other: &Self) -> bool {
+        self.data_source == other.data_source
+            && self.federated_security == other.federated_security
+            && self.user_id == other.user_id
+            && self.user_token == other.user_token
+            && self.password == other.password
+            && self.application_client_id == other.application_client_id
+            && self.application_key == other.application_key
+            && self.application_token == other.application_token
+            && self.application_certificate == other.application_certificate
+            && self.application_certificate_thumbprint == other.application_certificate_thumbprint
+            && self.authority_id == other.authority_id
+            && self.msi_auth == other.msi_auth
+            && self.msi_params == other.msi_params
+            && self.az_cli == other.az_cli
+    }
+}
+
+impl<'a> ConnectionString<'a> {
+    pub fn new(connection_string: &'a str) -> Result<Self, ConnectionStringError> {
+        let mut data_source = None;
+        let mut federated_security = None;
+        let mut user_id = None;
+        let mut user_token = None;
+        let mut password = None;
+        let mut application_client_id = None;
+        let mut application_token = None;
+        let mut application_key = None;
+        let mut application_certificate = None;
+        let mut application_certificate_thumbprint = None;
+        let mut authority_id = None;
+        let mut msi_auth = None;
+        let mut msi_params = None;
+        let mut az_cli = None;
+
+        let kv_str_pairs = connection_string
+            .split(';')
+            .filter(|s| !s.chars().all(char::is_whitespace));
+
+        for kv_pair_str in kv_str_pairs {
+            let mut kv = kv_pair_str.trim().split('=');
+            let k = match kv.next() {
```

Comment:
You can avoid the first match variant if you do `kv.next().filter(|k| !k.chars().all(char::is_whitespace))`

Reply:
nice, again saved a few lines.